### ISSUE TITLE
Conflicting dependencies with latest CRA

### ISSUE THREAD

**Issue by mikirobles** [JustFly1984/react-google-maps-api#150]
# Please provide an explanation

I'm getting an error trying to use this package with Create React App.
Seems like there's a lot of conflicting dependencies whose versions in this library are higher than those in CRA, thus triggering a preflight check error:

## Your Environment

os: mac

node --version v8.11.3

react version 16.8.6

## How is it behave?

When running `npm start` this error comes up:

```
There might be a problem with the project dependency tree.
It is likely not a bug in Create React App, but something you need to fix locally.

The react-scripts package provided by Create React App requires a dependency:

  "jest": "24.7.1"

Don't try to install it manually: your package manager does it automatically.
However, a different version of jest was detected higher up in the tree:

  /Users/******/dev/myproyect/node_modules/jest (version: 24.8.0) 

Manually installing incompatible versions is known to cause hard-to-debug issues.

If you would prefer to ignore this check, add SKIP_PREFLIGHT_CHECK=true to an .env file in your project.
That will permanently disable this message but you might encounter other issues.

To fix the dependency tree, try following the steps below in the exact order:

  1. Delete package-lock.json (not package.json!) and/or yarn.lock in your project folder.
  2. Delete node_modules in your project folder.
  3. Remove "jest" from dependencies and/or devDependencies in the package.json file in your project folder.
  4. Run npm install or yarn, depending on the package manager you use.

In most cases, this should be enough to fix the problem.
If this has not helped, there are a few other things you can try:

  5. If you used npm, install yarn (http://yarnpkg.com/) and repeat the above steps with it instead.
     This may help because npm has known issues with package hoisting which may get resolved in future versions.

  6. Check if /Users/micaelrobles/dev/myproyect/node_modules/jest is outside your project directory.
     For example, you might have accidentally installed something in your home folder.

  7. Try running npm ls jest in your project folder.
     This will tell you which other package (apart from the expected react-scripts) installed jest.

If nothing else helps, add SKIP_PREFLIGHT_CHECK=true to an .env file in your project.
That would permanently disable this preflight check in case you want to proceed anyway.
```

This is because @react-google-maps/api is installing a newer version of `jest` and so it makes this preflight check fail:

```
npm ls jest
employers@0.1.0 /Users/******/dev/myproyect
├─┬ @react-google-maps/api@1.2.7
│ └── jest@24.8.0 // <- this one is higher, so it conflicts with the react-scripts one down below
└─┬ react-scripts@3.0.1
  └── jest@24.7.1 
```

## How should it behave correctly?

It should let me run the project without having to disable preflight check.

Perhaps some of these dependencies should be `devDependencies`?

**Comment by JustFly1984** [comment]
@mikirobles Please use “resolve” property in your package.json

**Comment by mikirobles** [comment]
@JustFly1984 Sorry, do you mean "resolutions"?
https://yarnpkg.com/en/docs/selective-version-resolutions

**Comment by JustFly1984** [comment]
@mikirobles sorry, yes you are correct.

**Comment by erikhuisman** [comment]
eslint is listed in devDepencies AND dependencies in `package.json`

**Comment by mikirobles** [comment]
@erikhuisman right, actually *most* dependencies are in both `dependencies` and `devDependencies`. This doesn't seem right 🤔
.

**Comment by JustFly1984** [comment]
@erikhuisman @mikirobles Thank you for pointing this up! nice catch. I have fixed it in 1.3.0, and published. Enjoy!

**Comment by mikirobles** [comment]
@JustFly1984 Thank you very much!
